source/Calamari/HealthCheckCommand.cs

```diff
@@ -34,14 +36,21 @@ public Task Execute(RunningDeployment context)
             return ConfirmWebAppExists(account, resourceGroupName, webAppName);
         }
 
-        async Task ConfirmWebAppExists(ServicePrincipalAccount servicePrincipal, string resourceGroupName, string siteAndSlotName)
+        private async Task ConfirmWebAppExists(ServicePrincipalAccount servicePrincipal, string resourceGroupName, string siteAndSlotName)
         {
-            var azureClient = servicePrincipal.CreateAzureClient(); 
-            var webApp = await azureClient.WebApps.GetByResourceGroupAsync(resourceGroupName, siteAndSlotName);
-            if (webApp == null)
+            var client = servicePrincipal.CreateArmClient();
+            var subscription = await client.GetDefaultSubscriptionAsync();
+            var resourceGroups = subscription.GetResourceGroups();
+
+            try
+            {
+                ResourceGroupResource resourceGroup = await resourceGroups.GetAsync(resourceGroupName);
+                _ = await resourceGroup.GetWebSiteAsync(siteAndSlotName);
+            }
+            catch (RequestFailedException rfe) when (rfe.Status == 404)
```

Comment:
It looks now we only fail the health check if the exception is `RequestFailedException` with a `404` status code. Is this stricter than it was, and could possibly cause previously failing checks to become passing (i.e. a `401` was returned)?

Reply:
No, I don't think it will be stricter than it was. 

In the old version of the SDK, a Website not found was indicated by a `null` return value. However, the new SDK throws an exception in that case, which is why I replaced the null check with the try/catch. I intentionally scoped it down to 404, because that's specifically a "not found".

In both the old SDK and the new, any other failures are represented by the relevant exceptio. These aren't really "couldn't find the resource", they're other problems, so I don't want to control the Exception message, I just want them to just be printed raw into the deployment log as an error. An example of the auth failure scenario:

**Auth failure before change**
![image](https://user-images.githubusercontent.com/730704/173730763-ee2749af-b640-458d-b0d9-5f221df1d3b7.png)

**Auth failure after change**
![image](https://user-images.githubusercontent.com/730704/173731178-d65c43ce-5354-4514-b2ec-7a3968a5dcd7.png)

The main difference here is that I've now added more information into the controlled exception by wrapping the RequestFailedException, so under the new healthcheck, the failure message in the log will be longer as it will include the InnerException.

Reply:
Thanks for the explanation!